game.cpp

```diff
@@ -1222,31 +1220,22 @@ void test_game() //!OCLINT tests may be many
       eat_nth_food(g,0); // throws exception
     }
     catch ( const std::exception& e ) {
-      assert(e.what() == "You cannot eat food that already has been eaten!");
+      assert(std::string(e.what()) == std::string("You cannot eat food that already has been eaten!"));
```

Comment:
it's orbitrary, but this sort of test is testing if there are food to eat, and not testing the fact if the same food item is being eaten. Confusing